ls1intum/Artemis
# `Development`: Add PostgreSQL and test container support and cleanup change logs with migration paths

### Checklist
#### General


#### Server

### Motivation and Context
* We want to get rid of the many change logs, therefore there is a new initial schema
* We want to support PostgreSQL because it has certain advantages compared to MySQL

- [x] There are still some issues with PostgreSQL in some queries due to the use of native MySQL syntax or datatypes. All occurrences with timestamps / dates might need to be cast and `substring` on dates must be cast explicitly to string
- [x] Check the primary keys of the database: They might have changed between Postgres and Mysql -> Provide SQL script to migrate data

### Steps for Testing
By default, the tests are executed using `H2` with an improved mode (taking liquibase changes into account) and a bit faster than previously.
If you want to use PostgreSQL, you have to set the `postgres` profile, e.g. by running:

`SPRING_PROFILES_INCLUDE=postgres ./gradlew executeTests -x webapp`

If you want to use MySQL, you have to set the `mysql` profile, e.g. by running 

`SPRING_PROFILES_INCLUDE=mysql ./gradlew executeTests -x webapp`


Please navigate through all views of the Artemis app when running PostgreSQL and check if you notice Internal Server Errors  due to SQL syntax.

Example config for testing using Postgres in IntelliJ (which allows to debug)

<img width="1337" alt="image" src="https://user-images.githubusercontent.com/744067/209398694-44bc9dc0-e76a-46db-ab93-322a68f0629f.png">

Using MySQL:

<img width="1337" alt="image" src="https://user-images.githubusercontent.com/744067/209399783-facaf0cd-53c7-481a-9c82-543fe1cc0414.png">


### Review Progress





#### Code Review
#### Manual Tests
